components/plan-build/bin/public.bash

```diff
@@ -492,13 +498,14 @@ fix_interpreter() {
 pkg_interpreter_for() {
     local pkg=$1
     local int=$2
-    local path=$(_pkg_path_for_deps $pkg)
-    if [[ -z $path || $? -ne 0 ]]; then
+    local path
+    if ! path=$(_pkg_path_for_deps "$pkg") || [[ -z $path ]]; then
```

Comment:
Similar to https://github.com/habitat-sh/habitat/pull/4974/files#r192905861

Reply:
`$path` doesn't need to be quoted?

Reply:
No, because of the `[[` rather than `[`. Same situation as https://github.com/habitat-sh/habitat/pull/4974#discussion_r194919521